src/track_func.h

```diff
@@ -692,4 +692,25 @@ static inline bool IsUphillTrackdir(Slope slope, Trackdir dir)
 	return HasBit(_uphill_trackdirs[RemoveHalftileSlope(slope)], dir);
 }
 
+/**
+ * Determine the side in which the vehicle will leave the tile
+ *
+ * @param direction vehicle direction
+ * @param track vehicle track bits
+ * @return side of tile the vehicle will leave
+ */
+static inline DiagDirection VehicleExitDir(Direction direction, TrackBits track)
```

Comment:
I would rename this function, and leave vehicles out. Also, creating and extern table in rail.cpp should be considered, as it fits best with the rest of the set of functions involving tracks and dirs.

Reply:
I was going to but... uh... I couldn't come up with a name! Any ideas?

Reply:
YetAnotherFunctionWithTracksAndDirectionsReturningADirectionWhichIsInFactADiagDirection would do.

Looking the other related functions, I would name it TrackDirectionToExitdir(Track, Direction), but it is an ugly long name and you would need to convert the TrackBits to a single Track.

You could alternativelly use TrackdirToExitdir(TrackDirectionToTrackdir(track, dir)).